### PR TITLE
[stable/docker-registry] add token authentication

### PR DESCRIPTION
**Pull request by trollr**
Signed-off-by: Dennis Fricke <dennis.fricke@ninjaneers.de>

#### What this PR does / why we need it:

Adds token authentication 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by rendhalver** [comment]
Can you please sign off the second commit.

File: stable/docker-registry/templates/configmap.yaml [hunk 1]
```diff
@@ -10,3 +10,11 @@ metadata:
 data:
   config.yml: |-
 {{ toYaml .Values.configData | indent 4 }}
+{{- if .Values.auth.token.enabled }}
```
Comment (rendhalver) [inline stable/docker-registry/templates/configmap.yaml:13]: You can easily just add these values into your local values file.
The entire config for docker-registry  in in the values file.

Comment (trollr) [reply stable/docker-registry/templates/configmap.yaml:13]: Yep, but I need the rootcertbundle as a volumeMount. For the sake of completeness I added all token auth values and set a fix rootcertbundle path

Comment (rendhalver) [reply stable/docker-registry/templates/configmap.yaml:13]: What needs to access that?

Comment (alpire) [reply stable/docker-registry/templates/configmap.yaml:13]: the rootcertbundle is necessary to implement token based authentication. The registry uses the certificate to verify the token authenticity. I think this PR can be simplified. Users just need a way to mount a certificate in the container. Would you be open to a PR adding this capability?

**Pull request closed without merging**